### DISCUSSION TITLE
Restricting Encounters to Last 6 Years in FHIR Bundles

### DISCUSSION THREAD

**Discussion by mobashshir005** [synthetichealth/synthea#1502] [General]
Currently generated FHIR bundles have Encounter service data distribute across 100 years, so not able to get enough data for last 6 years. is there a way to generate records having Encounters only in last 6 years.

**Comment by hadleynet** [comment]
This will do that: `--exporter.years_of_history=6 --generate.only_alive_patients=true`

**Reply by mobashshir005** [reply]
@hadleynet Thank you for the quick response. Unfortunately, this still generates data for service periods like 1993 and 1997. I need to generate data only for service periods within the last 6 years, starting from the current year (2024).

**Reply by jawalonoski** [reply]
From the documentation, https://github.com/synthetichealth/synthea/wiki/Common-Configuration#other-export-settings, the setting `exporter.years_of_history`:

> The number of years of patient history to include in patient records. For example, if set to 5, then all patient history older than 5 years old (from the time you execute the program) will not be included in the exported records. Note that conditions and medications that are currently active will still be exported, regardless of this setting. Set this to 0 to keep all history in the patient record.

If you have data from 1993 and so forth, it is because the record contains data that is still _active_ today.

If you want to discard data that is still clinically relevant, you'll have to post-process it.

**Reply by mobashshir005** [reply]
Understood, Thank you  so much.
